runtime/src/accounts_index.rs

```diff
@@ -159,15 +159,31 @@ impl<'a, T: 'a + Clone> AccountsIndex<T> {
         reclaims: &mut SlotList<T>,
     ) -> Option<T> {
         if let Some(lock) = self.account_maps.get(pubkey) {
-            let mut list = &mut lock.1.write().unwrap();
-            // filter out other dirty entries
-            reclaims.extend(list.iter().filter(|(f, _)| *f == slot).cloned());
-            list.retain(|(f, _)| *f != slot);
-
-            lock.0.fetch_add(1, Ordering::Relaxed);
+            let list = &mut lock.1.write().unwrap();
+            // filter out other dirty entries from the same slot
+            let mut same_slot_previous_updates: Vec<(usize, (Slot, T))> = list
+                .iter()
+                .enumerate()
+                .filter_map(|(i, (f, value))| {
```

Comment:
nits: rename `f` => `s`? (That `f` stood for the `fork`).

Reply:
sure thing, updated!